### ISSUE TITLE
Suggest simplifying generic attribute names

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#67222]
**Brief description:**

If you write `[ProducesAttribute]`, diagnostic `IDE0001` suggests you simplify the name to `[Produces]`.  Could it be generalized to handle generic attributes?  That is, could it suggest simplifying `[ProducesAttribute<int>]` to `[Produces<int>]`?

**Languages applicable:**

This would be C# only since, as far as I know, VB doesn't allow generic attribute types.

**Code example that the analyzer should report:**

```csharp
public class ProducesAttribute : Attribute { }

public class ProducesAttribute<T> : Attribute { }

[ProducesAttribute] // "Attribute" grayed out
[ProducesAttribute<int>] // "Attribute" not grayed out
class C { }
```

**Additional information:**

[This](https://github.com/dotnet/roslyn/blob/aecd49800750d64e08767836e2678ffa62a4647f/src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs#L201) appears to be the code for handling the non-generic case.  Superficially, it looks like it should work, so maybe the fixer is choking?  I'm not sure which governs the fading.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by Rekkonnect** [comment]
Since this has not gotten attention for a while, may I take it up? I've got a fix for this.

**Comment by CyrusNajmabadi** [comment]
@Rekkonnect yes. Go ahead.
